substrate/frame/revive/rpc/src/example.rs

```diff
@@ -177,20 +203,77 @@ impl<Client: EthRpcClient + Send + Sync> TransactionBuilder<Client> {
 			.with_context(|| "Failed to fetch gas estimate")?;
 
 		println!("Gas estimate: {gas:?}");
-		let mut unsigned_tx = TransactionLegacyUnsigned {
-			gas,
-			nonce,
-			to,
-			value,
-			input,
-			gas_price,
-			chain_id,
-			..Default::default()
-		};
 
+		let mut unsigned_tx: TransactionUnsigned = match tx_type {
+			TransactionType::Legacy => TransactionLegacyUnsigned {
+				gas,
+				nonce,
+				to,
+				value,
+				input,
+				gas_price,
+				chain_id: Some(chain_id),
+				..Default::default()
+			}
+			.into(),
+			TransactionType::Eip2930 => Transaction2930Unsigned {
+				gas,
+				nonce,
+				to,
+				value,
+				input,
+				gas_price,
+				chain_id,
+				access_list: vec![],
+				r#type: TypeEip2930,
+			}
+			.into(),
+			TransactionType::Eip1559 => {
+				// For EIP-1559, we use gas_price as max_fee_per_gas and set
```

Comment:
I guess it doesn't really matter but since we don't support it set it to 0 for now